epytope/Core/Peptide.py

```diff
@@ -85,6 +85,22 @@ def __repr__(self):
             lines.append("in PROTEIN: %s" % p_id)
         return '\n'.join(lines)
 
+    """
+    Determines whether the :class:`~epytope.Core.Peptide.Peptide` is generated from a variant in the corresponding :class:`~epytope.Core.Protein.Protein`.
+
+    :return: True if peptide originates from a variant
+    :rtype: boolean
+    """
+    def is_created_by_variant(peptide):
+        transcript_ids = [x.transcript_id for x in set(peptide.get_all_transcripts())]
+        for t in transcript_ids:
+            prot = peptide.proteins[t]
+            for start_pos in peptide.proteinPos[t]:
+                for i in range(start_pos, start_pos + len(peptide)):
```

Comment:
Is all the traversing needed here? Could an alternative be something like this?:
`prots = self.get_all_proteins()`
`for prot in prots: if self in prot.vars.keys() return True else False`?

Reply:
Yesterday I did not see any easer solution to do this. `pro.vars.keys()` contains the protein position of the variants(https://github.com/KohlbacherLab/epytope/blob/86586c996a6b578e1be9fc10281720267688a6b8/epytope/Core/Protein.py#L51). 

This is basically a simplified version of https://github.com/KohlbacherLab/epytope/blob/86586c996a6b578e1be9fc10281720267688a6b8/epytope/Core/Peptide.py#L143:

Reply:
The idea here is to avoid usage of `get_variants_by_protein` since it is slower and returns ALL variants of the peptide, with the new function we only want to determine if there is at least ONE variant. 

Yes, the function will be used in the nf-core pipeline, for now I have included the method in the pipeline.

Reply:
Okay, makes sense!